### PR TITLE
Fix update date template based on date source

### LINKED ISSUE #1971: Date "according to Exif data" is sometimes wrong
In https://commons.wikimedia.org/wiki/File:Taiwan_white_caterpillar2018.jpg
you will observe "Date	1 November 2018 (according to Exif data)".
But if you scroll down to Metadata, you will find that is not the case.
App version: noted already on the page.

Please use the actual exif data date, not the upload date.
Nowhere in the exif data does it contain the upload date.

### PR DESCRIPTION
**Pull request by maskaravivek**
**Description (required)**

Fixes #1971

What changes did you make and why?

**Tests performed (required)**

Tested on ProdDebug. 

Google photo image(or any image without EXIF date): https://commons.wikimedia.org/wiki/File:Excavation_site_of_Ashokan_Pillar.jpg

Gallery image: https://commons.wikimedia.org/wiki/File:Baba_Black_Sheep_Varanasi.jpg

Edit: I have added [metadata-extractor](https://github.com/drewnoakes/metadata-extractor) library to reliably extract the file creation date from EXIF. This library can be reused for our other EXIF needs and also for IPTC.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/contributions/Contribution.java [hunk 1]
```diff
@@ -178,6 +194,24 @@ public String getPageContents(Context applicationContext) {
         return buffer.toString();
     }
 
+    private String getTemplatizedCreatedDate() {
```
Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/contributions/Contribution.java]: Could we get Javadocs please?

File: app/src/main/java/fr/free/nrw/commons/filepicker/UploadableFile.java [hunk 2]
```diff
@@ -80,18 +105,66 @@ public long getFileCreatedDate(Context context) {
...
+        return null;
+    }
+
     @Override
     public void writeToParcel(Parcel parcel, int i) {
         parcel.writeParcelable(contentUri, 0);
         parcel.writeSerializable(file);
     }
+
+    public class DateTimeWithSource {
```
Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/filepicker/UploadableFile.java]: Javadocs please

File: app/build.gradle [hunk 3]
```diff
@@ -27,7 +27,8 @@ dependencies {
     implementation 'com.jakewharton.rxbinding2:rxbinding-appcompat-v7:2.1.1'
     implementation 'com.jakewharton.rxbinding2:rxbinding-design:2.1.1'
     implementation 'com.facebook.fresco:fresco:1.10.0'
-    
+    implementation 'com.drewnoakes:metadata-extractor:2.11.0'
```
Comment (misaochan) [inline app/build.gradle]: Could you elaborate on the need for this library?

Comment (maskaravivek) [reply app/build.gradle]: @misaochan I have just updated the PR description to explain why I have added the library.

**Pull request merged**